### PR TITLE
refactor(gogoanime): Refactor fetchEpisodeSources

### PR DESCRIPTION
**Pull request by hase0278**
**What kind of change does this PR introduce?**
Refactoring fetch episode sources for slightly faster direct download link generator. Basically just made sure to make download link point to updated domain since referer hostname = download hostname? Or is it not always the case? Because if it is not always the case please reject this PR.

### CODE REVIEW

File: src/providers/anime/gogoanime.ts [hunk 1]
```diff
@@ -476,7 +476,7 @@ class Gogoanime extends AnimeParser {
       throw new Error('Something went wrong. Please try again later.');
     }
   };
-  fetchDirectDownloadLink = async (downloadUrl: string, captchaToken: string?): Promise<{ source: string | undefined; link: string | undefined }[]> => {
+  fetchDirectDownloadLink = async (downloadUrl: string, captchaToken: string): Promise<{ source: string | undefined; link: string | undefined }[]> => {
```
Comment (riimuru) [inline src/providers/anime/gogoanime.ts:479]: @hase0278 after removing the null safety on `captchaToken` the if statement on line 485 doesn't seem to be needed anymore

**Comment by riimuru** [comment]
I'm not sure if that's always the case with gogoanime download domain, but this is a good approach in case they change domains.

File: src/providers/anime/gogoanime.ts [hunk 2]
```diff
@@ -189,13 +189,13 @@ class Gogoanime extends AnimeParser {
               'User-Agent': USER_AGENT,
             },
             sources: await new StreamSB(this.proxyConfig, this.adapter).extract(serverUrl),
-            download: `https://gogohd.net/download${serverUrl.search}`,
+            download: `https://${serverUrl.host}t/download${serverUrl.search}`,
```
Comment (riimuru) [inline src/providers/anime/gogoanime.ts:192]: An unnecessary `t` after the hostname

File: src/providers/anime/gogoanime.ts (hunk 1 above)
Comment (hase0278) [reply src/providers/anime/gogoanime.ts:479]: Okay I can edit it if needed. Just been busy lately.

**Review by hase0278 — COMMENTED** [review]
Ammended

**Review by riimuru — APPROVED** [review]

**Pull request merged**